### PR TITLE
Update AWS One-Click docs

### PR DESCRIPTION
**Pull request by jac**
Use AWS CloudFormation to quickly deploy Sourcegraph.
Progress on:
  - https://github.com/sourcegraph/sourcegraph/issues/43074

This PR updates the AWS One-Click docs with more information and links to the completed CloudFormation template 
See here for more:
- https://github.com/sourcegraph/deploy/pull/12

## Test plan
See details [here](https://github.com/sourcegraph/deploy/pull/12) - use the Launch Stack button in the docs page to open the CloudFormation dialog in the preferred region (must have a default VPC which has a subnet with `Auto Assign Public IP` enabled)

### CODE REVIEW

File: doc/index.md [hunk 1]
```diff
@@ -54,12 +54,11 @@ Sourcegraph is runnable in a variety of environments, from cloud to self-hosted
...
   </a>
   <!-- AWS One Click-->
-  <a class="btn-app btn" href="/admin/deploy/docker-compose/aws-oneclick">
+  <a class="btn-app btn" href="/admin/deploy/machine-images/aws-oneclick">
     <img alt="aws-logo" src="/assets/other-logos/aws-light.svg" class="theme-light-only" />
     <img alt="aws-logo" src="/assets/other-logos/aws-dark.svg" class="theme-dark-only" />
     <h3>AWS One-Click</h3>
-    <span class="badge badge-warning">Coming soon</span> 
-    <!-- <p>Deploy onto AWS in one click</p> -->
+    <p>Deploy onto AWS in one click</p>
```
Comment (abeatrix) [inline doc/index.md:61]: 🎉Should we move the AWS One-Click to the top of our recommendation list? @slimsag @jac @jdpleiness

**Review by abeatrix — APPROVED** [review]
Well done! So excited for this 🎉

**Review by jdpleiness — APPROVED** [review]
LGTM! 🚀

File: doc/index.md (hunk 1 above)
Comment (slimsag) [reply doc/index.md:61]: Yes, definitely.

I also suggest we remove the _other_ AWS option (instructions for launching from an image) and instead just link to those from the one-click page.

**Review by slimsag — APPROVED** [review]
🔥

**Review by slimsag — APPROVED** [review]

**Pull request merged**